heron/spi/src/java/com/twitter/heron/spi/utils/TMasterUtils.java

```diff
@@ -114,49 +114,51 @@ private static boolean sendGetRequest(URL endpoint, String command,
    */
   private static TopologyAPI.TopologyState getRuntimeTopologyState(
       String topologyName,
-      SchedulerStateManagerAdaptor statemgr) {
+      SchedulerStateManagerAdaptor statemgr) throws TMasterException {
     PhysicalPlans.PhysicalPlan plan = statemgr.getPhysicalPlan(topologyName);
 
     if (plan == null) {
-      LOG.log(Level.SEVERE, "Failed to get physical plan for topology {0}", topologyName);
-      return null;
+      String errMsg = String.format("Failed to get physical plan for topology '%s'", topologyName);
+      throw new TMasterException(errMsg);
     }
 
     return plan.getTopology().getState();
   }
 
-  public static boolean transitionTopologyState(String topologyName,
+  public static void transitionTopologyState(String topologyName,
                                                 TMasterCommand topologyStateControlCommand,
                                                 SchedulerStateManagerAdaptor statemgr,
                                                 TopologyAPI.TopologyState startState,
                                                 TopologyAPI.TopologyState expectedState,
-                                                NetworkUtils.TunnelConfig tunnelConfig) {
+                                                NetworkUtils.TunnelConfig tunnelConfig)
+  throws TMasterException {
     TopologyAPI.TopologyState state = TMasterUtils.getRuntimeTopologyState(topologyName, statemgr);
     if (state == null) {
-      LOG.severe("Topology still not initialized.");
-      return false;
+      throw new TMasterException(String.format(
+          "Topology '%s' is not initialized yet", topologyName));
     }
 
     if (state == expectedState) {
-      LOG.log(Level.SEVERE, "Topology {0} command received topology {1} but already in {2} state",
-          new Object[] {topologyStateControlCommand, topologyName, state});
-      return true;
+      String errMsg = String.format(
+          "Topology {0} command received topology {1} but already in {2} state",
+          topologyStateControlCommand, topologyName, state);
+      throw new TMasterException(errMsg);
```

Comment:
I am not sure about using `Preconditions.checkState`. I don't see any particular benefit we could have. Besides, if we use it here, does that mean we should use it elsewhere? Does that mean we should use Guava as much as possible?